lib/OpenQA/CacheService.pm

```diff
@@ -130,9 +130,16 @@ sub run {
     $ENV{MOJO_INACTIVITY_TIMEOUT} //= 300;
     $app->log->debug("Starting cache service: $0 @args");
     $app->defaults->{service_pid} = $$;
-
-    my $cmd_return_code = $app->start(@args);
-    return $app->exit_code // $cmd_return_code // 0;
+    my $e;
+    my $cmd_return_code;
+    do {
+        eval { $cmd_return_code = $app->start(@args) };
```

Comment:
```suggestion
        my $cmd_return_code = eval { $cmd_return_code = $app->start(@args) };
```

Reply:
Do we know what's expected with async responses here? [Mojolicious::startup](https://docs.mojolicious.org/Mojolicious#startup) doesn't seem to document that?

Reply:
I think `start` blocks for the whole time the service is running. We *likely* don't need to care about anything happening asynchronously.

Reply:
I don't see anything happing asynchronously here.

Reply:
It depends on what command is executed by `start`. By default there are no async commands included though.

Reply:
Should we be using [around_dispatch](https://docs.mojolicious.org/Mojolicious#around_dispatch) instead? That seems to be the "native" approach to exception handling

Reply:
I'm wondering whether invoking `start` a 2nd time will ensure the event loop is reset. If not, I suppose that would be a good idea to do to avoid events from the previous run being processed on the next `start` invocation.

Reply:
@kalikiana You might be confusing this with another exception handling issue. This one is during app server startup, when the server can't bind to the socket because the interfaces are not ready yet.

Reply:
> I'm wondering whether invoking `start` a 2nd time will ensure the event loop is reset. If not, I suppose that would be a good idea to do to avoid events from the previous run being processed on the next `start` invocation.

Good point, i think this is currently undefined behaviour.